sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResource.java

```diff
@@ -117,27 +181,96 @@
+          final int count = rs.getRowsCount();
+
+          String pageStateStr = null;
+          BytesValue pagingStateOut = rs.getPagingState();
+          if (pagingStateOut.isInitialized()) {
+            ByteString rawPS = pagingStateOut.getValue();
+            if (!rawPS.isEmpty()) {
+              byte[] b = rawPS.toByteArray();
+              pageStateStr = Base64.getEncoder().encodeToString(b);
+            }
+          }
+
+          Sgv2RowsResponse response = new Sgv2RowsResponse(count, pageStateStr, convertRows(rs));
+          return javax.ws.rs.core.Response.status(Response.Status.OK).entity(response).build();
+        });
+  }
+
+  @Timed
+  @POST
+  @ApiOperation(
+      value = "Add row",
+      notes =
+          "Add a row to a table in your database. If the new row has the same primary key as that of an existing row, the database processes it as an update to the existing row.",
+      response = String.class,
+      responseContainer = "Map",
+      code = 201)
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            code = 201,
+            message = "resource created",
+            response = Map.class,
+            responseContainer = "Map"),
+        @ApiResponse(code = 400, message = "Bad Request", response = RestServiceError.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = RestServiceError.class),
+        @ApiResponse(code = 409, message = "Conflict", response = RestServiceError.class),
+        @ApiResponse(
+            code = 500,
+            message = "Internal server error",
+            response = RestServiceError.class)
+      })
+  public Response createRow(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(value = "Name of the table to use for the request.", required = true)
+          @PathParam("tableName")
+          final String tableName,
+      @ApiParam(value = "", required = true) final String payload,
+      @Context HttpServletRequest request) {
+    if (isAuthTokenInvalid(token)) {
+      return invalidTokenFailure();
     }
+    final String cql = buildAddRowCQL(keyspaceName, tableName, payload);
+    logger.info("createRow(): try to call backend with CQL of '{}'", cql);
 
-    Sgv2RowsResponse response = new Sgv2RowsResponse(count, pageStateStr, convertRows(rs));
-    return javax.ws.rs.core.Response.status(Response.Status.OK).entity(response).build();
+    StargateGrpc.StargateBlockingStub blockingStub =
+        grpcFactory.constructBlockingStub().withCallCredentials(new StargateBearerToken(token));
```

Comment:
how much is this costing and should this be in some kind of cache?

Reply:
That would probably be one of the first optimization once we can measure performance, when we have end-to-end functionality (all operations for at least one successful test). Same goes for async handling.